library/elf/elf.c

```diff
@@ -71,6 +71,7 @@ void elf_load(int pid, elf_reader reader, int argc, void** argv) {
 
     earth->mmu_alloc(&ppage_id, &base);
     earth->mmu_map(pid, args_start++, ppage_id);
+    memset((char*)base, 0, PAGE_SIZE);
```

Comment:
Would you please explain why this is necessary?

I feel like, if we do `proc_set[curr_proc_idx].syscall.msg.received = PENDING` in `excp_entry` as mentioned above, we will not need this `memset`.

Reply:
It was necessary because the previous contents of the user syscall 					    memory had nonzero `syscall.msg.received` on the ARTY Board, which made `SYS_RECV` automatically succeed. Your suggestions solved this issue.